crates/ruff_python_formatter/src/statement/stmt_match.rs

```diff
@@ -1,12 +1,55 @@
-use crate::{not_yet_implemented, FormatNodeRule, PyFormatter};
-use ruff_formatter::{write, Buffer, FormatResult};
+use ruff_formatter::{format_args, write, Buffer, FormatResult};
 use ruff_python_ast::StmtMatch;
 
+use crate::comments::dangling_comments;
+use crate::expression::maybe_parenthesize_expression;
+use crate::expression::parentheses::Parenthesize;
+use crate::not_yet_implemented_custom_text;
+use crate::prelude::*;
+use crate::{FormatNodeRule, PyFormatter};
+
 #[derive(Default)]
 pub struct FormatStmtMatch;
 
 impl FormatNodeRule<StmtMatch> for FormatStmtMatch {
     fn fmt_fields(&self, item: &StmtMatch, f: &mut PyFormatter) -> FormatResult<()> {
-        write!(f, [not_yet_implemented(item)])
+        let StmtMatch {
+            range: _,
+            subject,
+            cases,
+        } = item;
+
+        let comments = f.context().comments().clone();
+        let dangling_item_comments = comments.dangling_comments(item);
+
+        write!(
+            f,
+            [
+                text("match"),
+                space(),
+                maybe_parenthesize_expression(subject, item, Parenthesize::IfBreaks),
+                text(":"),
+                dangling_comments(dangling_item_comments)
+            ]
+        )?;
+
+        for case in cases {
+            write!(
+                f,
+                [block_indent(&format_args![
+                    text("case"),
+                    space(),
+                    not_yet_implemented_custom_text("NOT_YET_IMPLEMENTED_MatchCase"),
+                    text(":"),
+                    block_indent(&case.body.format())
+                ])]
+            )?;
```

Comment:
Could we move the mock implementations to the match cases instead and call into `case.format().fmt(f)` here? 

This has the benefit that implementing a specific match case requires no changes to the statement formatting (and we can even provide match case mocks of the right pattern)

Reply:
Yeah, we can.

> (and we can even provide match case mocks of the right pattern)

By this, do you mean the different kinds of patterns for the match cases? If so, that will make the formatting unstable because in the first iteration each pattern type will be formatted as it's not implemented equivalent but on the second iteration (now that the pattern is a variable semantically), some of them will be formatted again.

```python
match p:
	case Point(x, y):  # Pattern::MatchClass
		pass

# vvvvv
match p:
	case NOT_YET_IMPLEMENTED_MatchClass:  # Pattern::MatchValue
		pass

# vvvvv
match p:
	case NOT_YET_IMPLEMENTED_MatchValue:
		pass
```

Reply:
You can use the not_implemented_custom_text to provide a specific text. You then need to make sure that the class item uses a class verbatim text.

If it's not already the case: The MatchCase formatting should be a match statement that dispatches the formatting to the specific case item (see FormatStmt)